capsules/src/fm25cl.rs

```diff
@@ -48,8 +49,8 @@ pub trait FM25CLClient {
     fn done(&self, buffer: &'static mut [u8]);
 }
 
-pub struct FM25CL<'a> {
-    spi: &'a hil::spi::SPIMasterDevice,
+pub struct FM25CL<'a, S: hil::spi::SpiMasterDevice + 'a> {
+    spi: &'a S,
```

Comment:
Ah, I see. Technically, you could write something like:

```rust
pub struct FM25CL<'a> {
    spi: &'a hil::spi::SpiMasterDevice<ChipSelect=usize>,
   ...
```

but it does mean you have to specify what type the `ChipSelect` is here, which you don't wanna do since you don't know a-priori what kind of SPI controller the FM25CL will be connected to (you could parameterize the FM25CL with the ChipSelect type, but at that point, what you're doing is probably better).